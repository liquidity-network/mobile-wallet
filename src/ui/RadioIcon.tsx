import React, { PureComponent } from 'react'
import Ionicons from 'react-native-vector-icons/Ionicons'

import { ACTIVE_COLOR, GREY } from 'globalStyles'

interface Props {
  enabled?: boolean
  color?: string
}

export default class RadioIcon extends PureComponent<Props> {
  render() {
    return (
      <Ionicons
        name={this.props.enabled ? 'md-radio-button-on' : 'ios-radio-button-off'}
        size={22}
        color={this.props.enabled ? this.props.color || ACTIVE_COLOR : GREY}
      />
    )
  }
}
