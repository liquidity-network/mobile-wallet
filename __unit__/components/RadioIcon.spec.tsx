import React from 'react'
import { create } from 'react-test-renderer'

import RadioIcon from 'ui/RadioIcon'

describe('<RadioIcon> component', () => {
  it('renders correctly', () => {
    const tree = create(<RadioIcon enabled={false} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders enabled correctly', () => {
    const tree = create(<RadioIcon enabled />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
