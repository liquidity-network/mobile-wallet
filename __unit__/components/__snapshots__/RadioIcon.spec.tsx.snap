// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioIcon> component renders correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#9fa6b3",
        "fontSize": 22,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`<RadioIcon> component renders enabled correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#5436cd",
        "fontSize": 22,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
